pkg/cvo/cvo.go

```diff
@@ -495,3 +503,23 @@ func (b *resourceBuilder) Apply(ctx context.Context, m *lib.Manifest) error {
 	}
 	return builder.Do(ctx)
 }
+
+func hasNeverReachedLevel(cv *configv1.ClusterVersion) bool {
```

Comment:
This is a stretch :P (sorry my brain just went there)
but, what if somebody overrides the desired more than 10 times, so there will be all partial and we will truncate to all partial because https://github.com/openshift/cluster-version-operator/blob/2a78a6cedd73f5f0bbfbeb8778e23d66490d7ea7/pkg/cvo/status.go#L74-L76

Reply:
Hrm, we should leave the last completed.  We should never prune to no completed.  Good catch, will change